android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java

```diff
@@ -36,21 +60,41 @@ protected Result doInBackground(Void... ignored) {
     }
 
     Result result = null;
+    /**
+     * mCameraViewWidth and mCameraViewHeight are obtained from portait orientation
+     * mWidth and mHeight are measured with landscape orientation with Home button to the right
+     * adjustedCamViewWidth is the adjusted width fromt the Aspect ratio setting 
+     */
+    int adjustedCamViewWidth = (int) (mCameraViewHeight / mRatio);
+    float adjustedScanY = (((adjustedCamViewWidth - mCameraViewWidth) / 2) + (mScanAreaY * mCameraViewWidth)) / adjustedCamViewWidth;
+    
+    int left = (int) (mScanAreaX * mWidth);
+    int top = (int) (adjustedScanY * mHeight);
+    int scanWidth = (int) (mScanAreaWidth * mWidth);
+    int scanHeight = (int) (((mScanAreaHeight * mCameraViewWidth) / adjustedCamViewWidth) * mHeight);
 
     try {
       BinaryBitmap bitmap = generateBitmapFromImageData(
               mImageData,
               mWidth,
               mHeight,
-              false
+              false,
+              left,
+              top,
+              scanWidth,
+              scanHeight
       );
       result = mMultiFormatReader.decodeWithState(bitmap);
     } catch (NotFoundException e) {
```

Comment:
@bqwang91 any idea why this exception is raised? thanks a lot

Reply:
@bqwang91 I believe you are raising this exception to try scan the barcode in landscape mode and I believe it does not work in landscape as discussed in the pull request comments.. I willl comment that code for now and come back to it later. Thanks

Reply:
https://github.com/zxing/zxing/issues/891

Reply:
[the following logic](https://github.com/bqwang91/react-native-camera/blob/583507f22363d330f903af0215f20250a7ee3851/android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java#L89-L98) works when using the backcamera. The functionality does not work with the frontcamera. I did not test your example yet using the frontcamera, did you implement the logic for the frontcamera or is it included in one of the other try statements? I believe the image should not be rotated with the frontcamera...